airflow-core/tests/unit/always/test_secrets.py

```diff
@@ -119,6 +120,38 @@ def test_backend_fallback_to_env_var(self, mock_get_connection):
 
         assert conn.get_uri() == "mysql://airflow:airflow@host:5432/airflow"
 
+    @conf_vars(
+        {
+            (
+                "secrets",
+                "backend",
+            ): "airflow.providers.amazon.aws.secrets.systems_manager.SystemsManagerParameterStoreBackend",
+            ("secrets", "backend_kwargs"): '{"connections_prefix": "/airflow", "profile_name": null}',
+            ("secrets", "backends_order"): "custom,environment_variable,metastore",
+        }
+    )
+    def test_backends_order(self):
+        backends = ensure_secrets_loaded()
+        backend_classes = [backend.__class__.__name__ for backend in backends]
+        assert backend_classes == [
+            "SystemsManagerParameterStoreBackend",
+            "EnvironmentVariablesBackend",
+            "MetastoreBackend",
+        ]
+
+    @pytest.mark.parametrize(
+        "backends_order",
+        [
+            pytest.param("custom,metastore", id="no_environment_variable_backend"),
+            pytest.param("environment_variable", id="no_metastore_backend"),
+            pytest.param("metastore,environment_variable,unsupported", id="unsupported_backend"),
+        ],
+    )
+    def test_backends_order_invalid_cases(self, backends_order):
+        with conf_vars({("secrets", "backends_order"): backends_order}):
+            with pytest.raises(AirflowConfigException):
+                ensure_secrets_loaded()
```

Comment:
Needs more test coverage to cover the cases for worker mode too. This only checks for "non workers"

Reply:
Added